public/js/listeners.js

```diff
@@ -135,7 +135,7 @@ function startListeners() {
             }
         }
 
-        if (e.keyCode == 8 || e.key == "Delete") {
+        if (simulationArea.lastSelected && (e.keyCode == 8 || e.key == "Delete")) {
```

Comment:
This is wrong because `simulationArea.lastSelected` could be undefined but `simulationArea.multipleObjectSelections` needn't be an empty list.

Reply:
@satu0king, I did this because :

* if the user wants to add a new circuit using combination analysis then there is no need to select previous elements.
* If Something is selected outside the simulation area then it should deselect the elements.

Should I remove the line ??

Reply:
Undo the changes to this line `if (simulationArea.lastSelected && (e.keyCode == 8 || e.key == "Delete")) {`

These changes are correct
```
    simulationArea.multipleObjectSelections = [];
    simulationArea.lastSelected = undefined;
```

Reply:
@satu0king if i change that line back to ` if (e.keyCode == 8 || e.key == "Delete") ` the backspace will not work due to the line `$("input").blur();` inside `delete_selected()` as the current iput will lose it focus.